src/Api.Rest.Tests/Dtos/Data/SearchConditionParserTest.cs

```diff
@@ -0,0 +1,202 @@
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2021                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Api.Rest.Tests.Dtos.Data
+{
+	#region usings
+
+	using System;
+	using System.Collections.Generic;
+	using FluentAssertions;
+	using NUnit.Framework;
+	using Zeiss.PiWeb.Api.Rest.Dtos.Data;
+
+	#endregion
+
+	[TestFixture]
+	public class SearchConditionParserTest
+	{
+		#region members
+
+		private static IEnumerable<( string condition, bool result )> _TestCases = new[]
+		{
+			("123=[]", true),
+			("123=]", false),
+			("123=[", false),
+			("123=condition", false),
+			("123=[condition", false),
+			("123=condition]", false),
+			("123=[condit]ion", false),
+			("123=[condit]ion]", true),
+			("123=[condition[]]", true),
+			("123=[]condition]", true),
+			("123=[con[]dition]", true),
+			("123=[con[]dition]+124=[con[]dition]", true),
+			("123=[con[]dition]+124=[con[]dition", false)
+		};
+
+		#endregion
+
+		#region methods
+
+		[Test]
+		public void Parse_ConditionContainsBrackets_Success()
+		{
+			var expected = new GenericSearchAttributeConditionDto()
+			{
+				Attribute = 123,
+				Operation = OperationDto.Equal,
+				Value = "condition[with[b[]rackets"
+			};
+
+			var actual = SearchConditionParser.Parse( "123=[condition[with[b[]rackets]" );
+
+			actual.Should().BeEquivalentTo( expected );
+		}
+
+		[Test]
+		public void Parse_MultiConditionContainsBrackets_Success()
+		{
+			var expected = new GenericSearchAndDto()
+			{
+				Conditions = new GenericSearchConditionDto[]
+				{
+					new GenericSearchAttributeConditionDto()
+					{
+						Attribute = 123,
+						Operation = OperationDto.Equal,
+						Value = "condition[]with[]b[ack]ets"
+					},
+					new GenericSearchAttributeConditionDto()
+					{
+						Attribute = 124,
+						Operation = OperationDto.In,
+						Value = "[100;-1;23];[2;-12;40];[-22;33;44]"
+					}
+				}
+			};
+
+			var actual = (GenericSearchAndDto)SearchConditionParser.Parse( "123=[condition[]with[]b[ack]ets]+124In[[100;-1;23];[2;-12;40];[-22;33;44]]" );
+
+			// ReSharper disable once CoVariantArrayConversion
+			actual.Conditions.Should().BeEquivalentTo( expected.Conditions );
+		}
+
+		[Test]
+		public void Parse_MissingClosingBracket_ThrowsException()
+		{
+			Action action = () => SearchConditionParser.Parse( "123=[value" );
+			action.Should().Throw<InvalidOperationException>().Where( e => e.Message.Contains( "']'" ) );
+		}
+
+		[Test]
+		public void Parse_MissingOpeningBracket_ThrowsException()
+		{
+			Action action = () => SearchConditionParser.Parse( "123=value]" );
+			action.Should().Throw<InvalidOperationException>().Where( e => e.Message.Contains( "[]" ) );
+		}
+
+		[Test]
+		[TestCaseSource( nameof( _TestCases ) )]
+		public void Parse_CorrectSearchConditions_DoesNotThrow( (string, bool) testcase )
```

Comment:
The name implies that given test cases are correct but given test cases seems to can invalid. Can you adapt the test method?

Reply:
Done